src/wallet/wallet.cpp

```diff
@@ -1731,7 +1732,16 @@ CAmount CWallet::GetColdStakingBalance() const
 
 CAmount CWallet::GetStakingBalance(const bool fIncludeColdStaking) const
 {
-    return GetBalance() + (fIncludeColdStaking ? GetColdStakingBalance() : 0);
+    AssertLockHeld(cs_main);
```

Comment:
This shouldn't be here, it should be the opposite `AssertNotLockHeld`.

The `cs_main` lock is being acquired inside the `loopTxsBalance`.

Reply:
Right. Didn't notice `loopTxsBalance` already acquired the lock.
Fixed